birdhouse/read-configs.include.sh

```diff
@@ -0,0 +1,131 @@
+#!/bin/sh
+# This script is meant to be sourced by all scripts needing to read default.env
+# or env.local.
+#
+# Normally only read_configs() is needed, but for any special needs, each step
+# can be called individually.
+#
+# When not using read_configs(), caller is responsible for knowing the proper
+# sequence to read the config files, for hiding password when reading env.local
+# and to call process_delayed_eval() to have fully usable resolved variable
+# values.
+
+# Derive COMPOSE_DIR from the most probable locations.
+# This is NOT meant to be exhautive.
+# Caller of this file can simply set COMPOSE_DIR itself.
+discover_compose_dir() {
+    if [ -z "$COMPOSE_DIR" -o ! -e "$COMPOSE_DIR" ]; then
+        if [ -e "./pavics-compose.sh" ]; then
+            # Current dir is COMPOSE_DIR
+            COMPOSE_DIR="`realpath .`"
+        elif [ -e "../pavics-compose.sh" ]; then
+            # Parent dir is COMPOSE_DIR
+            # Case of all the scripts under deployment/ or scripts/
+            COMPOSE_DIR="`realpath ..`"
+        elif [ -e "../birdhouse-deploy/birdhouse/pavics-compose.sh" ]; then
+            # Case of sibling checkout at same level as birdhouse-deploy.
+            COMPOSE_DIR="`realpath "../birdhouse-deploy/birdhouse"`"
+        elif [ -e "../../birdhouse-deploy/birdhouse/pavics-compose.sh" ]; then
+            # Case of subdir of sibling checkout at same level as birdhouse-deploy.
+            COMPOSE_DIR="`realpath "../../birdhouse-deploy/birdhouse"`"
+        elif [ -e "../../../birdhouse-deploy/birdhouse/pavics-compose.sh" ]; then
+            # Case of sub-subdir of sibling checkout at same level as birdhouse-deploy.
+            COMPOSE_DIR="`realpath "../../../birdhouse-deploy/birdhouse"`"
+        elif [ -e "./birdhouse/pavics-compose.sh" ]; then
+            # Child dir is COMPOSE_DIR
+            COMPOSE_DIR="`realpath birdhouse`"
+        fi
```

Comment:
For readability, a for loop seems a better to try/break various options.

Also, is there a technical (older shell support?) reason to use ``` `` ``` instead of `$()`? Using a shell-checker on this file flags those calls. `$()` is better as it allows nesting calls and are more readable.

Reply:
A for-loop to try/break various options, could you give an example?  I am not sure how to write a for-loop in this case.

And yes I used backticks instead of `$()` for portability reason.  I try to stay within the limited features of the original Bourne shell (/bin/sh) and not Bourne Again shell (/bin/bash).

Now that you asked, I tried to search for differences between sh and bash and I could not find anything definitive which syntax is which shell.  So to play safe, I'd rather stay with backticks for now if you don't mind.

Reply:
Something similar to this:
```shell
TESTS="
  test1
  ../test2
  ../../test3
  ./test4
"
for path in ${TESTS}; do
  if [ -d "${path}" ]; then
    echo "Found ${path}"
    COMPOSE_DIR="${path}"
    break
  fi
done

```

Reply:
`$()` are ok in `sh`
I'm using it (and have to because it's nested) here: 
https://github.com/bird-house/birdhouse-deploy/blob/master/birdhouse/scripts/get-components-json.include.sh#L57-L63

Reply:
I switched to `$()` except in `process_delayed_eval` since I could not get the `eval` to work properly.  Looks like in some corner cases backticks still have its use.

As for the for-loop try/catch, it's not applicable to this case because there is a different mapping for each case:
```
./pavics-compose.sh -> realpath .
../pavics-compose.sh -> realpath ..
./birdhouse/pavics-compose.sh -> realpath birdhouse
../birdhouse-deploy/birdhouse/pavics-compose.sh -> realpath ../birdhouse-deploy/birdhouse
...
```